### PR TITLE
feat(#38): migrate to the screen system

### PR DESCRIPTION
**Pull request by aizensoosuke**

### CODE REVIEW

File: app/src/main/java/com/github/studydistractor/sdp/ui/HistoryScreen.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.github.studydistractor.sdp.ui
+
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.runtime.*
+import com.github.studydistractor.sdp.history.HistoryInterface
+import com.github.studydistractor.sdp.ui.components.MessageCard
+
+/**
+ * Author: Bluedrack
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/ui/HistoryScreen.kt:10]: You can rewrite the comment from the activity :
Screen showing the historic of the current logged user fetching data online using the HistoryInterface.

**Review by bluedrack — CHANGES_REQUESTED** [review]
There is some minor issues. But globally everything seems to work properly.

File: app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt
Comment (lipefree) [inline app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt:1]: ?_?

File: app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt
Comment (lipefree) [inline app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: What are you injecting in this activity ?

**Review by lipefree — CHANGES_REQUESTED** [review]
Overall good PR, there is just too much useless comments in some random places. Except from few comments, this really make the app looks wayyyyy nicer !

File: app/src/main/java/com/github/studydistractor/sdp/StudyDistractorScreen.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            composable(route = StudyDistractorScreen.CreateDistraction.name)  {
+                CreateDistractionScreen()
+            }
+            composable(route = StudyDistractorScreen.History.name) {
+                HistoryScreen(hi = FirebaseHistory())
+            }
+        }
+    }
+
+    val authenticated: Boolean = true;
```
[the remark this answers is not in the document]
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/StudyDistractorScreen.kt:123]: Yes ! I do need to remove them. Thanks

File: app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt [hunk 3]
```diff
@@ -31,9 +31,9 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
     var locationPermissionGranted = false
     var map: GoogleMap? = null
     private var lastKnownLocation: Location? = null
-    private val defaultLocation = LatLng(46.520544, 6.567825)
+    private val defaultLocation = LatLng(46.520544, 6.567825) // Done
```
[the remark this answers is not in the document]
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt:34]: Definitely useless

File: app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt [hunk 4]
```diff
@@ -31,9 +31,9 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
...
 
-    object Constants {
+    object Constants { // DONE
```
[the remark this answers is not in the document]
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt:36]: What is it with // Done in my code, I don't even remember those

File: app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt [hunk 5]
```diff
@@ -152,7 +152,7 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
      * Gets the current location of the device, and positions the map's camera.
      */
     @SuppressLint("MissingPermission")
-    private fun getDeviceLocation() {
+    private fun getDeviceLocation() { // DOING
```
[the remark this answers is not in the document]
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/maps/MapsActivity.kt:155]: wut

File: app/src/main/java/com/github/studydistractor/sdp/ui/HistoryScreen.kt (hunk 1 above)
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/ui/HistoryScreen.kt:10]: Yes sorry I missed that

File: app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt
Comment (aizensoosuke) [reply app/src/androidTest/java/com/github/studydistractor/sdp/sillysillycoverage/SillySillyCoverageTest.kt:1]: We need 80% coverage -_--_--_-

File: app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: Actually, nothing, does it mean I should remove Hilt_MainActivity ?

Comment (lipefree) [reply app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: Yes and also @AndroidEntryPoint which is indicating to Hilt in which class to inject !

Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: I see ! Sorry for the rookie mistake

Comment (lipefree) [reply app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: But you should be inheriting from appcompactactivity

Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/MainActivity.kt:1]: Done ! Thanks :)

**Review by lipefree — APPROVED** [review]
LGTM now !

**Review by bluedrack — APPROVED** [review]
LGTM. Everything seems to work properly. Nice work.

**Pull request merged**